pkg/trace/obfuscate/sql.go

```diff
@@ -309,8 +312,9 @@ func (f *tableFinderFilter) Reset() {
 
 // ObfuscatedQuery specifies information about an obfuscated SQL query.
 type ObfuscatedQuery struct {
-	Query     string // the obfuscated SQL query
-	TablesCSV string // comma-separated list of tables that the query addresses
+	Query     string      // the obfuscated SQL query
+	TablesCSV string      // comma-separated list of tables that the query addresses
+	Metadata  SQLMetadata // metadata extracted from the SQL query
```

Comment:
Since tables is a top-level attribute, does it make sense to make `Comments` also a top-level attribute?

Reply:
Good question, imo, I think it's a nicer interface having a separate metadata attribute to hold everything rather than making all the metadata we collect a top-level attribute. My reason for that is we can simply inject the metadata type into the different filters to collect whatever we need at that stage in a single interface. It _could_ get messy in the long run if we continue to expand the various metadata types we collect.

e.g
```
metadata        SQLMetadata
discard         = discardFilter{metadata: &metadata}
replace         = replaceFilter{metadata: &metadata, replaceDigits: opts.ReplaceDigits}
grouping        groupingFilter{metadata: &metadata}
ableFinder     = tableFinderFilter{metadata: &metadata, storeTableNames: storeTableNames}
```

But something I neglected was that `Tables` should be placed under metadata if we proceed with this pattern.